### PR TITLE
Add Ledger support

### PR DESCRIPTION
**Pull request by bartosz-lipinski**
Changes:
1. Added support for logout
2. Add separate way to auth using ledger
3. Add WalletProvider interface that supports localStorage and ledger
4. Enable HTTPS for local development

TODO: 
- [x] Error handling
- [x] Large transaction signing
- [ ] Expose non-root ledger accounts
- [x] Code refactor (move ledger logic to single folder)

UX preview: 
image

### CODE REVIEW

**Comment by bartosz-lipinski** [comment]
For SPL transactions we will need migrate to new format (transfer2, approve2, mintTo2). Solana ledger app doesn't recognized old format and requires user to enable 'blind sign'. I will submit separate PR for that.

**Comment by nishadsingh1** [comment]
Awesome! Thanks for putting this together

Will review in the coming days

**Review by sconybeare — CHANGES_REQUESTED** [review]
Can you split the refactoring and the ledger support changes into separate PRs? Currently it's difficult to review the changes for correctness.

**Comment by bartosz-lipinski** [comment]
> Can you split the refactoring and the ledger support changes into separate PRs? Currently it's difficult to review the changes for correctness.

@sconybeare not sure what you mean by refactoring changes everything in this PR was need to get ledger up and running.

File: src/pages/LoginPage.js [hunk 1]
```diff
@@ -17,21 +16,29 @@ import FormControlLabel from '@material-ui/core/FormControlLabel';
...
-          <Link style={{ cursor: 'pointer' }} onClick={() => setRestore(true)}>
-            Restore existing wallet
-          </Link>
+          <div style={{ display: 'flex', flexDirection: 'row' }}>
+            <Button color="primary" 
+                    onClick={() => login('ledger')}>
+              Use Ledger
+            </Button>
+
+            <Button type="secondary" style={{ cursor: 'pointer', marginLeft: 'auto', marginRight: 0, textTransform: 'none', fontWeight: 'normal' }}
```
Comment (nathanielparke) [inline src/pages/LoginPage.js:37]: can we run prettier to correct indents here?

File: src/components/NavigationFrame.js [hunk 2]
```diff
@@ -177,16 +179,24 @@ function WalletSelector() {
...
           }}
         >
           <ListItemIcon className={classes.menuItemIcon}>
             <AddIcon fontSize="small" />
           </ListItemIcon>
           Create Account
+        </MenuItem>}
+
+        <MenuItem
+          onClick={() => {
```
Comment (nathanielparke) [inline src/components/NavigationFrame.js:195]: this should be 
```
onClick={() => {
  setAnchorEl(null);
  logout();
}}
```

null is un-necessary for the logout function, and setting the dropdown state to null when we logout fixes the dropdown rendering opened in a weird place when we log back in

**Review by nathanielparke — COMMENTED** [review]
Left a few comments, but mostly looks good.

**Pull request closed without merging**